.github/workflows/release.yml

```diff
@@ -0,0 +1,26 @@
+on:
+  release:
+    types: [published]
+    
+jobs:
+  pypi-publish:
+    name: Upload release to PyPI
+    runs-on: ubuntu-latest
+    environment:
+      name: pypi
+      url: https://pypi.org/p/tubular
+    permissions:
+      id-token: write  # IMPORTANT: this permission is mandatory for trusted publishing
+    steps:
+    - uses: actions/checkout@v4
+    - uses: actions/setup-python@v4
+      with:
+        python-version: "3.8"
+    - name: "Build Package"
+      run: |
+        python setup.py sdist bdist_wheel
+        pip install twine 
+        twine check dist/* 
+
```

Comment:
Thought: I think that we're missing a stage that uploads the binary and source packages to the releases page. From looking at the [softprops/action-gh-release](https://github.com/softprops/action-gh-release#%EF%B8%8F-uploading-release-assets) action, I think that it's as simple as adding:

```
- name: Github Release
        uses: softprops/action-gh-release@v1
        with:
          files: dist/*
```

Reply:
I ahve set this up to run on the event that we publish to GH. At the moment that process publishes the source code on GH and then this triggered pipeline would automatically build and upload binaries to pypi.

I think if we want to use the above action to publish a release to github too we would not be able to use publication to GH as the trigger.

From looking into it I think manually trigegring the GH action as described [here](https://docs.github.com/en/actions/using-workflows/manually-running-a-workflow) and publishing to both from this workflow is probably a better approach.

Reply:
changed [here](https://github.com/lvgig/tubular/pull/161/commits/37297c1083bbfa29b86bfb9c20d4862855257afe)

Reply:
looks like this action had the capability to upload assets to a published release but is now [ archived ](https://github.com/actions/upload-release-asset)

Reply:
I do like where this is going - it's the protected branch only (so main once we go with #136). And, whilst I can see the sense of always creating the artefacts upon manual runs on main, do we really want every manually triggered run to publish to PyPI? Might be worth a `if tag` level to the release? Or am I misunderstanding the intention?

Reply:
I think this may become a bit of a chicken and egg situation as the GH release creates the tagged branch.  This also raises the issue that the tag name needs to be set.  By default the [GH release action](https://github.com/softprops/action-gh-release#%EF%B8%8F-uploading-release-assets) uses the [github.ref variable](https://docs.github.com/en/actions/learn-github-actions/variables#default-environment-variables) but I'm not sure yet how to set that as the version number.

I'm tempted to remove the GH release from this action and add that in later as a separate feature.  Either going back to triggering on a GH release or a manual run of the workflow like we have currently but with an if tag block for the release portion.

Reply:
Yeah, I get your point. Shall we remove the `Github Release` stage and go back to the `on: release` trigger? This work item relates to publishing to PyPI, so let's get that working.

Reply:
Reverted to triggering on publish and removed publish to GH step [here](https://github.com/lvgig/tubular/pull/161/commits/1e54ea1278ee49acd7f26532fb1b4e6ee09ea0db)